sdk/kms/src/aws/aws-kms-signer.ts

```diff
@@ -0,0 +1,147 @@
+	 */
+	getKeyScheme() {
+		return SIGNATURE_FLAG_TO_SCHEME[this.#publicKey.flag() as SignatureFlag];
+	}
+
+	/**
+	 * Retrieves the public key associated with this signer.
+	 * @returns The Secp256k1PublicKey instance.
+	 * @throws Will throw an error if the public key has not been initialized.
+	 */
+	getPublicKey() {
+		return this.#publicKey;
+	}
+
+	/**
+	 * Signs the given data using AWS KMS.
+	 * @param bytes - The data to be signed as a Uint8Array.
+	 * @returns A promise that resolves to the signature as a Uint8Array.
+	 * @throws Will throw an error if the public key is not initialized or if signing fails.
+	 */
+	async sign(bytes: Uint8Array): Promise<Uint8Array> {
+		const signResponse = await this.#client.runCommand('Sign', {
+			KeyId: this.#kmsKeyId,
+			Message: toBase64(bytes),
+			MessageType: 'RAW',
+			SigningAlgorithm: 'ECDSA_SHA_256',
+		});
+
+		// Concatenate the signature components into a compact form
+		return this.#getConcatenatedSignature(fromBase64(signResponse.Signature));
+	}
+
+	/**
+	 * Synchronous signing is not supported by AWS KMS.
+	 * @throws Always throws an error indicating synchronous signing is unsupported.
+	 */
+	signData(): never {
+		throw new Error('KMS Signer does not support sync signing');
+	}
+
+	/**
+	 * Generates a concatenated signature from a DER-encoded signature.
+	 *
+	 * This signature format is consumable by Sui's `toSerializedSignature` method.
+	 *
+	 * @param signature - A `Uint8Array` representing the DER-encoded signature.
+	 * @returns A `Uint8Array` containing the concatenated signature in compact form.
+	 *
+	 * @throws {Error} If the input signature is invalid or cannot be processed.
+	 */
+	#getConcatenatedSignature(signature: Uint8Array): Uint8Array {
+		if (!signature || signature.length === 0) {
+			throw new Error('Invalid signature');
+		}
+
+		// Initialize a DERElement to parse the DER-encoded signature
+		const derElement = new DERElement();
+		derElement.fromBytes(signature);
+
+		const [r, s] = derElement.toJSON() as [string, string];
+
+		switch (this.getKeyScheme()) {
+			case 'Secp256k1':
+				return new secp256k1.Signature(BigInt(r), BigInt(s)).normalizeS().toCompactRawBytes();
+			case 'Secp256r1':
+				return new secp256r1.Signature(BigInt(r), BigInt(s)).normalizeS().toCompactRawBytes();
+		}
+
+		// Create a Secp256k1Signature using the extracted r and s values
+		const secp256k1Signature = new secp256k1.Signature(BigInt(r), BigInt(s));
+
+		// Normalize the signature and convert it to compact raw bytes
+		return secp256k1Signature.normalizeS().toCompactRawBytes();
+	}
+
+	/**
+	 * Prepares the signer by fetching and setting the public key from AWS KMS.
+	 * It is recommended to initialize an `AwsKmsSigner` instance using this function.
+	 * @returns A promise that resolves once a `AwsKmsSigner` instance is prepared (public key is set).
+	 */
+	static async fromCredentials(keyId: string, options: AwsClientOptions) {
```

Comment:
fromCredentials name seems a bit off to me, maybe something like `fromKeyId` or something similar might be better?

Reply:
yeah, "Key ID" is the primary argument here, makes sense to use that name